tests/Mantis/PluginTest.php

```diff
@@ -0,0 +1,151 @@
+
+/**
+ * Helper API tests
+ * @package Tests
+ * @subpackage String
+ */
+class MantisPluginTest extends PHPUnit_Framework_TestCase {
+
+	const MANTISCORE = 'MantisCore';
+
+	/**
+	 * References of version requirements to test against MantisCore versions.
+	 */
+	const REQ_13 = '1.3';
+	const REQ_130 = '1.3.0';
+	const REQ_131 = '1.3.1';
+	const REQ_13_20 = '1.3, <2.0';
+	const REQ_13_21 = '1.3, <2.1';
+	const REQ_13_30 = '1.3, <3.0';
+	const REQ_20 = '2.0';
+	const REQ_21 = '2.1';
+	const REQ_210 = '2.1.0';
+	const REQ_210_DEV = '2.1.0-dev';
+	const REQ_210_A1 = '2.1.0-beta.1';
+	const REQ_210_B1 = '2.1.0-beta.1';
+	const REQ_211 = '2.1.1';
+
+	/**
+	 * Tests setup.
+	 */
+	public function setUp() {
+		plugin_register(self::MANTISCORE);
+	}
+
+
+	/**
+	 * Tests plugin_dependency() for MantisCore.
+	 *
+	 * @param string $p_core_version MantisCore version to test against
+	 * @param string $p_requirement  Version requirement string
+	 * @param bool   $p_expected     True if dependency OK (function returned 1)
+	 *
+	 * @dataProvider providerDependency
+	 */
+	public function testPluginDependency( $p_core_version, $p_requirement, $p_expected ) {
+		global $g_plugin_cache;
+
+		# Set MantisCore version for the test
+		$g_plugin_cache[self::MANTISCORE]->version = $p_core_version;
+
+		$t_result = plugin_dependency(
+			self::MANTISCORE,
+			$p_requirement
+		);
+
+		$this->assertEquals(
+			$p_expected,
+			$t_result == 1,
+			"MantisCore '$p_core_version' should "
+			. ( $p_expected ? 'meet' : 'have failed' )
+			. " requirement '$p_requirement'"
+		);
+	}
+
+	/**
+	 * Provides a series of dependency test cases.
+	 *
+	 * Test case structure:
+	 *   array( <core version>, <requirement>, <result> )
+	 *
+	 * plugin_dependency() is called to check the given <core version> against
+	 * <requirement>. <result> indicates whether we expect the test to
+	 * pass (i.e. function returns 1) or fail (returns value is 0 or -1).
+	 *
+	 * @return array List of test cases
+	 */
+	public function providerDependency() {
+		return array(
+			array( '1.3.0', self::REQ_13, true ),
+			array( '1.3.0', self::REQ_130, true ),
+			array( '1.3.0', self::REQ_131, false ),
+			array( '1.3.0', self::REQ_13_20, true ),
+			array( '1.3.0', self::REQ_13_30, true ),
+			array( '1.3.0', self::REQ_20, false ),
+
+			array( '2.0.0', self::REQ_13, false ),
+			array( '2.0.0', self::REQ_13_20, false ),
+			array( '2.0.0', self::REQ_13_21, true ),
+			array( '2.0.0', self::REQ_13_30, true ),
+			array( '2.0.0', self::REQ_20, true ),
+			array( '2.0.0', self::REQ_210_DEV, false ),
+			array( '2.0.0-beta.1', self::REQ_20, true ),
+
+			array( '2.1.0', self::REQ_13, false ),
+			array( '2.1.0', self::REQ_13_20, false ),
+			array( '2.1.0', self::REQ_13_21, false ),
+			array( '2.1.0', self::REQ_13_30, true ),
+			array( '2.1.0', self::REQ_20, true ),
+			array( '2.1.0', self::REQ_21, true ),
+			array( '2.1.0', self::REQ_210_DEV, true ),
+			array( '2.1.0', self::REQ_210_B1, true ),
```

Comment:
Often such unit tests are good documentation for expected behavior, so even if mechanically it doesn't add value, it may provide a better documentation for how versions relate to each other.  But you can argue semver.org provides that in this case.